stateproof/worker_test.go

```diff
@@ -108,31 +110,33 @@ func (s *testWorkerStubs) addBlock(ccNextRound basics.Round) {
 }
 
 func (s *testWorkerStubs) StateProofKeys(rnd basics.Round) (out []account.StateProofRecordForRound) {
-	for _, part := range s.keys {
-		if part.OverlapsInterval(rnd, rnd) {
-			partRecord := account.ParticipationRecord{
-				ParticipationID:   part.ID(),
-				Account:           part.Parent,
-				FirstValid:        part.FirstValid,
-				LastValid:         part.LastValid,
-				KeyDilution:       part.KeyDilution,
-				LastVote:          0,
-				LastBlockProposal: 0,
-				LastStateProof:    0,
-				EffectiveFirst:    0,
-				EffectiveLast:     0,
-				VRF:               part.VRF,
-				Voting:            part.Voting,
-			}
-			signerInRound := part.StateProofSecrets.GetSigner(uint64(rnd))
-			partRecordForRound := account.StateProofRecordForRound{
-				ParticipationRecord: partRecord,
-				StateProofSecrets:   signerInRound,
-			}
-			out = append(out, partRecordForRound)
+	return stateProofRecordFromKeys(rnd, s.keys)
+}
+
+func stateProofRecordFromKeys(rnd basics.Round, keys []account.Participation) (out []account.StateProofRecordForRound) {
```

Comment:
Why have you created this function? You feed it with the same keys as you have in the `testWorkerStubs` in the test below. why not call `s.stateProofRecordFromKeys(rnd)` instead?

Reply:
Because I do not have the object `s` `testWorkerStubs`.